jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -663,6 +722,35 @@ ecma_op_object_put (ecma_object_t *object_p, /**< the object */
       }
       break;
     }
+    case ECMA_OBJECT_TYPE_CLASS:
+    {
+#ifndef CONFIG_DISABLE_TYPEDARRAY_BUILTIN
+      /* ES2015 9.4.5.5 */
+      if (ecma_is_typedarray (ecma_make_object_value (object_p)))
+      {
+        ecma_number_t num = ecma_string_to_number (property_name_p);
+        ecma_string_t *num_to_str = ecma_new_ecma_string_from_number (num);
```

Comment:
In JerryScript, strings which are also valid array indicies (e.g. "34", "0", "64567") are stored as uint32 numbers to reduce space and improve array operations. For example, in case of `a[i]` the 'i' must be converted to string first according to the standard. This "conversion" is cheap if `i` is an uint32 number or a string (these two are the most frequent cases).

The uint32 "strings" are also represent the canonical forms of each decimal integer number between 0 and UINT32_MAX. Since we will not support array buffers bigger than UINT32_MAX, only these uint32 "strings" can be valid array indicies. All other numbers are non integer numbers or not in canonical form, so they can be rejected (return with undefined) without decoding their number values.

So the pseudo code would look like this:

    if (ECMA_STRING_GET_CONTAINER (string_p) == ECMA_STRING_CONTAINER_UINT32_IN_DESC)
      return get_value(string_p->u.uint32_number)
    else if (is_string_is_number(string_p))
      return undefined
    else
      normal_code_path

The only question is what happens with "-0.0"